Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -2670,6 +2669,25 @@ void RenderStyle::setNonInheritedCustomPropertyValue(const AtomString& name, Ref
     m_rareNonInheritedData.access().customProperties.access().setCustomPropertyValue(name, WTFMove(value));
 }
 
+const CSSCustomPropertyValue* RenderStyle::customPropertyValue(const AtomString& name, const Style::CustomPropertyRegistry& registry) const
+{
+    if (auto* value = customPropertyValueWithoutResolvingInitial(name))
+        return value;
+
+    // Alternatively we could just initialize the registered initial values to each RenderStyle.
+    auto* registered = registry.get(name);
+    return registered ? registered->initialValue.get() : nullptr;
+}
+
+const CSSCustomPropertyValue* RenderStyle::customPropertyValueWithoutResolvingInitial(const AtomString& name) const
+{
+    for (auto* map : { &nonInheritedCustomProperties(), &inheritedCustomProperties() }) {
+        if (auto* value = map->get(name))
+            return value;
+    }
+    return nullptr;
+}
```

Comment:
Maybe this could be a single method where the `CustomPropertyRegistry` is a pointer and you only check for an initial value if the registry is provided?

Reply:
I had that initially but I don't want people wondering what happens if they don't provide a registry (and they always should).